server/src/gql/resolver.py

```diff
@@ -72,3 +72,66 @@ def resolve_add_new_mates(
     family.mate_ids += [new_mate.id for new_mate in new_mates]
     family.save()
     return new_mates
+
+
+@mutation.field("assignMatesToFamilies")
+def resolve_assign_mates_to_families(
+    obj: Any, info: GraphQLResolveInfo, newAssignments: List[NewAssignmentInput]
+) -> List[str]:
+    """
+    Assign mates to specified families.
+
+    Args:
+    ----
+        obj: GraphQL
+        info: GraphQLInfo
+        newAssignments: each item of NewAssignmentInput contains
+            - familyId
+            - list of id's of mates to be assigned to that family
+
+    Returns ids of updated mates
+
+    """
+    user = User.lookup_user(current_user.get_id())
+    if not user:
+        raise Exception("User not found")
+
+    moved_mates: List[str] = []
+
+    unassigned_family = user.unassigned_family
+    unassigned_id = str(unassigned_family.id)
+    unassigned_mate_ids = set([str(mate.id) for mate in unassigned_family.mate_ids])
+    family_ids = set([str(family.id) for family in user.family_ids])
+    for input in newAssignments:
+        for mate_id in input["mateIds"]:
+            if mate_id not in unassigned_mate_ids:
+                raise Exception(f"Mate id {mate_id} is not unassigned")
+        if input["familyId"] not in family_ids:
+            raise Exception(f"Family id {input['familyId']} does not exist")
+
+    families_map = {str(family.id): family for family in user.families}
+    for input in newAssignments:
+        for mate_id in input["mateIds"]:
+            idx = next(
+                idx
+                for idx in range(len(families_map[unassigned_id].mate_ids))
+                if str(families_map[unassigned_id].mate_ids[idx].id) == mate_id
+            )
+            mate_ref = families_map[unassigned_id].mate_ids.pop(idx)
+            families_map[input["familyId"]].mate_ids.append(mate_ref)
+            moved_mates.append(str(mate_ref.id))
```

Comment:
Imo, we dont have to concern the backend with the mate being assigned to two different families in the same call. The frontend should just give good input - worst case, mate gets assigned to the wrong place - which seems like an ok failure case.